static/app/utils/useMutateApiToken.tsx

```diff
@@ -0,0 +1,86 @@
+import type {InternalAppApiToken} from 'sentry/types/user';
+import {
+  getApiQueryData,
+  setApiQueryData,
+  useMutation,
+  useQueryClient,
+} from 'sentry/utils/queryClient';
+import type RequestError from 'sentry/utils/requestError/requestError';
+import useApi from 'sentry/utils/useApi';
+
+const API_TOKEN_QUERY_KEY = ['/api-tokens/'] as const;
+
+type UpdateTokenQueryVariables = {
+  name: string;
+};
+type FetchApiTokenParameters = {
+  tokenId: string;
+};
+export const makeFetchApiTokenKey = ({tokenId}: FetchApiTokenParameters) =>
+  [`/api-tokens/${tokenId}/`] as const;
+
+interface UseMutateApiTokenProps {
+  token: InternalAppApiToken;
+  onError?: (error: RequestError) => void;
+  onSuccess?: () => void;
+}
+
+export default function useMutateApiToken({
+  token,
+  onSuccess,
+  onError,
+}: UseMutateApiTokenProps) {
+  const api = useApi();
+  const queryClient = useQueryClient();
+  return useMutation<{}, RequestError, UpdateTokenQueryVariables>({
```

Comment:
for `{}` type, that's the expected response type. will this return a new token or anything?

Reply:
this won't return anything, but the new token will be passed to the `onSuccess` method if the token is successfully modified